### PR TITLE
AV-91225 fix hostname extraction

### PR DESCRIPTION
**Pull request by ThejasNU**

### CODE REVIEW

**Comment by glennga** [comment]
can you run this using the templates/agents/with_controlflow README instructions? i'm getting a ValueError:

```
(my-agent-py3.12) ➜  with_controlflow git:(main) agentc publish tool prompt --bucket travel-sample
--------------------------------------------------------------------------------------------------------------------------------------
TOOL
--------------------------------------------------------------------------------------------------------------------------------------
Using the catalog identifier: c5082e32d45acdc2ac2056f3a67acb6df1ad2100

Uploading the tool catalog items to Couchbase.
find_routes_with_one_layover: 100%|████████████████████████████████████████████████████████████████████| 2/2 [00:00<00:00, 475.63it/s]
Tool catalog items successfully uploaded to Couchbase!

Now building the GSI indexes for the tool catalog.
v1_agent_catalog_toolcat_annotations: 100%|███████████████████████████████████████████████████████████| 4/4 [00:00<00:00, 1019.64it/s]
All GSI indexes for the tool catalog have been successfully created!

Now building the vector index for the tool catalog.
ERROR: Cannot convert given value of max partition to integer: int() argument must be a string, a bytes-like object or a real number, not 'NoneType'
Traceback (most recent call last):
  File "/Users/glenn.galvizo/Projects/AgentCatalog/libs/agentc_core/agentc_core/util/ddl.py", line 115, in create_vector_index
    max_partition_env = int(max_partition_env)
                        ^^^^^^^^^^^^^^^^^^^^^^
TypeError: int() argument must be a string, a bytes-like object or a real number, not 'NoneType'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/glenn.galvizo/Library/Caches/pypoetry/virtualenvs/my-agent-708wextc-py3.12/bin/agentc", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/Users/glenn.galvizo/Projects/AgentCatalog/libs/agentc/agentc/__main__.py", line 5, in main
    agentc_cli.main()
  File "/Users/glenn.galvizo/Projects/AgentCatalog/libs/agentc_cli/agentc_cli/main.py", line 833, in main
    raise e
  File "/Users/glenn.galvizo/Projects/AgentCatalog/libs/agentc_cli/agentc_cli/main.py", line 808, in main
    click_main()
  File "/Users/glenn.galvizo/Library/Caches/pypoetry/virtualenvs/my-agent-708wextc-py3.12/lib/python3.12/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/glenn.galvizo/Library/Caches/pypoetry/virtualenvs/my-agent-708wextc-py3.12/lib/python3.12/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/Users/glenn.galvizo/Library/Caches/pypoetry/virtualenvs/my-agent-708wextc-py3.12/lib/python3.12/site-packages/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/glenn.galvizo/Library/Caches/pypoetry/virtualenvs/my-agent-708wextc-py3.12/lib/python3.12/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/glenn.galvizo/Library/Caches/pypoetry/virtualenvs/my-agent-708wextc-py3.12/lib/python3.12/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/glenn.galvizo/Library/Caches/pypoetry/virtualenvs/my-agent-708wextc-py3.12/lib/python3.12/site-packages/click/decorators.py", line 33, in new_func
    return f(get_current_context(), *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/glenn.galvizo/Projects/AgentCatalog/libs/agentc_cli/agentc_cli/main.py", line 549, in publish
    cmd_publish(
  File "/Users/glenn.galvizo/Projects/AgentCatalog/libs/agentc_cli/agentc_cli/cmds/publish.py", line 169, in cmd_publish
    _, err = create_vector_index(bucket, k, connection_details_env, dims)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/glenn.galvizo/Projects/AgentCatalog/libs/agentc_core/agentc_core/util/ddl.py", line 117, in create_vector_index
    raise ValueError(f"Cannot convert given value of max partition to integer: {e}") from e
ValueError: Cannot convert given value of max partition to integer: int() argument must be a string, a bytes-like object or a real number, not 'NoneType'
```

**Comment by ThejasNU** [comment]
> provided

I will fix this issue and push the changes

**Comment by ThejasNU** [comment]
@glennga @TJ202 I have updated the environment variable handling, please re-review it

**Comment by TJ202** [comment]
> @glennga @TJ202 I have updated the environment variable handling, please re-review it

added a path for that: https://github.com/couchbaselabs/agent-catalog/pull/26/
this allows string as well int inputs (nishanth was facing an issue when he was giving int as value but string was being sent somehow so the patch handles both cases)

File: libs/agentc_core/agentc_core/util/ddl.py [hunk 1]
```diff
@@ -108,18 +112,19 @@ def create_vector_index(
 
     max_partition_env = os.getenv("AGENT_CATALOG_MAX_SOURCE_PARTITION")
     try:
-        max_partition_env = int(max_partition_env)
+        max_partition = int(max_partition_env) if max_partition_env is not None else 1024
```
Comment (TJ202) [inline libs/agentc_core/agentc_core/util/ddl.py:115]: ref to https://github.com/couchbaselabs/agent-catalog/pull/26/ for this - you can revert this change

File: libs/agentc_core/agentc_core/util/ddl.py [hunk 2]
```diff
@@ -108,18 +112,19 @@ def create_vector_index(
...
     except Exception as e:
-        raise ValueError(f"Cannot convert given value of max partition to integer: {e}") from e
+        raise ValueError(
+            f"Cannot convert given value of max source partition to integer: {e}\nUpdate the environment variable 'AGENT_CATALOG_MAX_SOURCE_PARTITION' to an integer value"
```
Comment (TJ202) [inline libs/agentc_core/agentc_core/util/ddl.py:118]: this is good - can merge into the patch

File: libs/agentc_core/agentc_core/util/ddl.py [hunk 3]
```diff
@@ -26,7 +26,7 @@ def is_index_present(
     auth = (conn.username, conn.password)
 
     # Make a request to FTS until a live node is reached. If all nodes are down, try the host.
-    for fts_node_hostname in fts_nodes_hostname + [conn.host]:
+    for fts_node_hostname in fts_nodes_hostname:
```
Comment (TJ202) [inline libs/agentc_core/agentc_core/util/ddl.py:29]: add conn.host back

**Review by TJ202 — CHANGES_REQUESTED** [review]

File: libs/agentc_core/agentc_core/util/ddl.py (hunk 3 above)
Comment (ThejasNU) [reply libs/agentc_core/agentc_core/util/ddl.py:29]: done

**Review by TJ202 — DISMISSED** [review]

**Review by TJ202 — APPROVED** [review]

**Pull request merged**